e2e/nomostest/clusters/kind.go

```diff
@@ -202,3 +204,48 @@ apiServer:
 	// We failed to create the cluster maxKindTries times, so fail out.
 	return err
 }
+
+var preloadImages = []string{
+	testing.HTTPDImage,
+	testing.RegistryImage,
+	testing.NginxImage,
+}
+
+// Pull each public image a single time on the host, and then sideload the images
+// into each kind cluster. This reduces the number of times each image is pulled
+// to avoid 429 too many requests errors from public docker registries.
+// Alternatively we may choose to host these images in a registry we control.
+func sideLoadImages(clusterName string) error {
+	if err := pullImages(); err != nil {
```

Comment:
How about extract pullImages to before creating kind cluster?

Reply:
I don't think it makes much difference if we pull before vs after the kind cluster is created. I changed it to pull in parallel to kind cluster creation as that could be an optimization.